base/sort.jl

```diff
@@ -82,16 +82,80 @@ issorted(itr;
     lt=isless, by=identity, rev::Bool=false, order::Ordering=Forward) =
     issorted(itr, ord(lt,by,rev,order))
 
-function select!(v::AbstractVector, k::Union{Int,OrdinalRange}, o::Ordering)
+function partialsort!(v::AbstractVector, k::Union{Int,OrdinalRange}, o::Ordering)
     inds = indices(v, 1)
     sort!(v, first(inds), last(inds), PartialQuickSort(k), o)
-    v[k]
+
+    if k isa Integer
```

Comment:
I think inlining is clearer because otherwise, you have to go look for `_dispatchftw` and you might not realize it is defined just below. But I am generally of the opinion that inlining stuff is to prefer over breaking out things in function, when those functions are only used in one place.